packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -77,6 +84,57 @@ abstract class TextSelectionControls {
 
   /// Returns the size of the selection handle.
   Size get handleSize;
+
+  void handleCut(TextSelectionDelegate delegate) {
+    Clipboard.setData(new ClipboardData(
```

Comment:
You use delegate.textEditingValue here a lot, I think you should assign it to a local final.